File: 0001.two_sum/Solution.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> dict = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int another = target - nums[i];
            if (null != dict.get(another)) {
                return new int[]{dict.get(another), i};
            }

            dict.put(nums[i], i);
        }

        return null;
    }
}

File: 0001.two_sum/solution.go
package solution

func twoSum(nums []int, target int) []int {
	dict := map[int]int{}

	for i := 0; i < len(nums); i++ {
		anothor := target - nums[i]
		if j, ok := dict[anothor]; ok {
			return []int{j, i}
		}

		dict[nums[i]] = i
	}

	return nil
}
